utils/debug/TileDebug.js

```diff
@@ -117,101 +116,81 @@ export default function createTileDebugUI(datDebugTool, view, layer, debugInstan
-                    }
+                    const obbHelper = new OBBHelper(node.obb);
+                    obbHelper.layer = layer;
+                    helper.add(obbHelper);
                 } else if (layer.id == sb_layer_id) {
                     const color = new THREE.Color(Math.random(), Math.random(), Math.random());
                     const material = new THREE.MeshBasicMaterial({ color: color.getHex(), wireframe: true });
-                    helper = new THREE.Mesh(geometrySphere, material);
-                    helper.position.copy(node.boundingSphere.center);
-                    helper.scale.multiplyScalar(node.boundingSphere.radius);
+                    const sphereHelper = new THREE.Mesh(geometrySphere, material);
+                    sphereHelper.position.copy(node.boundingSphere.center);
+                    sphereHelper.scale.multiplyScalar(node.boundingSphere.radius);
+                    sphereHelper.scale.set(1, 1, 1).multiplyScalar(node.boundingSphere.radius);
+                    sphereHelper.layer = layer;
+                    helper.add(sphereHelper);
                 }
 
-                helper.layers.set(l3js);
-                helper.layer = layer;
-                node.add(helper);
-                helper.updateMatrixWorld(true);
-
-                // if we don't do that, our OBBHelper will never get removed,
-                // because once a node is invisible, children are not removed
-                // any more
-                const hiddenHandler = node.material.addEventListener('hidden', () => {
-                    node.material.removeEventListener(hiddenHandler);
-                    let i = node.children.length;
-                    while (i--) {
-                        const c = node.children[i];
-                        if (c.layer === sb_layer_id) {
-                            if (c.dispose) {
-                                c.dispose();
-                            } else if (Array.isArray(c.material)) {
-                                for (const material of c.material) {
-                                    material.dispose();
-                                }
-                            } else {
-                                c.material.dispose();
-                            }
-                            node.children.splice(i, 1);
-                        }
-                    }
-                });
+                node.link.push(helper);
             }
 
-            if (layer.id == sb_layer_id) {
-                helper.position.copy(node.boundingSphere.center);
-                helper.scale.set(1, 1, 1).multiplyScalar(node.boundingSphere.radius);
-            }
+            layer.object3d.add(helper);
             helper.updateMatrixWorld(true);
-            helper.visible = true;
         } else if (helper) {
-            helper.visible = false;
+            layer.object3d.remove(helper);
+        }
+    }
+
+    class DebugLayer extends GeometryLayer {
```

Comment:
I think we could add it to a different file and use it in 3DTilesDebug.js too, WDYT?

Reply:
I will do it in a function to refactor the debug

Reply:
Sorry,  I'll open a new PR to refactor the debug

Reply:
ok :+1: